new/prusti/src/specs/typed.rs

```diff
@@ -47,6 +47,10 @@ impl StructuralToTyped<AssertionKind> for json::AssertionKind {
         use json::AssertionKind::*;
         match self {
             Expr(expr) => AssertionKind::Expr(expr.to_typed(typed_expressions)),
+            And(vec_assertions) => {
+                // ???
+                AssertionKind::And()
```

Comment:
The following should do the job:
```rust
AssertionKind::And(assertions.into_iter().map(|assertion| assertion.to_typed(typed_expressions))
```

Reply:
This fails with
```
expected struct `std::vec::Vec`, found struct `std::iter::Map`
```
Does it mean that I should change the enum field in `json::AssertionKind` accordingly?

Reply:
I forgot `.collect()` after `.map(...)`.